src/main/java/utils/Ui.java

```diff
@@ -0,0 +1,162 @@
+package utils;
+
+import task.Task;
+
+import java.util.List;
+
+/**
+ * A Ui class to manage all of Duke's output. It functions with a singleton
+ * model so that at any point in time, only one copy of the output String is
+ * being built.
+ */
+public class Ui {
+    public static final String LIST_ACTION_TITLE =
+            "Here are the tasks in your list:\n";
+    public static final String FIND_ACTION_TITLE =
+            "Here are the matching tasks in your list:\n";
+    public static final String SAD_EMOTICON = "\u2639"; // "☹"
+    private static final String LOGO =
+            " ____        _        \n"
+                    + "|  _ \\ _   _| | _____ \n"
+                    + "| | | | | | | |/ / _ \\\n"
+                    + "| |_| | |_| |   <  __/\n"
+                    + "|____/ \\__,_|_|\\_\\___|\n";
+
+    private static final String DIVIDER = "____________________________________________________\n";
+
+    private static StringBuilder outputBuilder;
+
+    /**
+     * Generates an instance object of the singleton class Ui to build
+     * the outputBuilder.
+     *
+     * @return An instance of Ui object.
+     */
+    public static Ui getInstance() {
+        return new Ui();
+    }
+
+    private Ui() {
+        if (outputBuilder == null) {
+            outputBuilder = new StringBuilder();
+        }
+    }
+
+    /**
+     * Prints a hello message when program first initializes.
+     */
+    public void printWelcomeMessage() {
+        outputBuilder.append("Hello from\n" + LOGO);
+        outputBuilder.append(DIVIDER);
+        outputBuilder.append("Hello! I'm Duke\n");
+        outputBuilder.append("What can I do for you?\n");
+        outputBuilder.append(DIVIDER);
+    }
+
+    /**
+     * Prints a list of actions that can be used.
+     */
+    public void printHelpMessage() {
```

Comment:
I like the help message function a lot! It serves a good purpose of introducing and assisting users of the functionality.